File: src/main/scala/coda/Context.scala
package coda

import com.oracle.truffle.api.{ Truffle, TruffleLanguage }

class CoreContext(val language: CoreLanguage, var env: TruffleLanguage.Env) {
  val singleThreadedAssumption = Truffle.getRuntime.createAssumption("context is single threaded")
  def shutdown: Unit = {}
}

File: src/main/java/stg/Context.java
package stg;

import com.oracle.truffle.api.Assumption;
import com.oracle.truffle.api.Truffle;
import com.oracle.truffle.api.TruffleLanguage;

public class Context {
  public Context(Language language, TruffleLanguage.Env env) {
    this.language = language;
    this.env = env;
  }
  public Language language;
  public TruffleLanguage.Env env;
  
  public final Assumption singleThreadedAssumption = Truffle.getRuntime().createAssumption("context is single threaded");

  public void shutdown() { }
}